include/jemalloc/internal/spin_delay_arm.h

```diff
@@ -0,0 +1,33 @@
+#include "jemalloc/internal/jemalloc_preamble.h"
+#include <sys/auxv.h>
+#include <stdatomic.h>
+
+/* Define HWCAP_SB if not already defined in system headers */
+#ifndef HWCAP_SB
+#define HWCAP_SB (1ULL << 56) /* Speculation Barrier */
+#endif   // HWCAP_SB
+
+/* Global variable to track SB support, initialized by constructor */
+static _Atomic int arm_has_sb_instruction = ATOMIC_VAR_INIT(0);
+
+/* Constructor function to detect hardware capabilities at program startup */
+__attribute__((constructor))
+static void
+detect_arm_sb_support(void) {
```

Comment:
Likewise, this is a `__attribute__((constructor)) static`, which exists separately in each TU. This should be defined in only one TU, again such as some `spin_delay_arm.c`.

There may be a non-`static` declaration of the function here. Likewise, I'm not sure if this function also has to be marked as hidden, or if somehow all symbols are automatically marked as hidden in some special linker script or build step.

Reply:
Alternatively, can also just use normal global variable initialization.

```spin_delay_arm.c
const int arm_has_sb_instruction = getauxval(AT_HWCAP) & HWCAP_SB);
```